server/webapp/WEB-INF/rails.new/app/assets/new_javascripts/models/agents/environments.js

```diff
@@ -18,17 +18,24 @@ define(['mithril', 'lodash', 'helpers/mrequest', 'models/agents/tri_state_checkb
   var Environments  = {};
   Environments.list = [];
 
+  var getSortedEnvironments = function (environments, selectedAgents) {
+    var sortedEnvironments = _.sortBy(environments);
+    var selectedAgentsEnvironments = _.map(selectedAgents, function (agent) {
+      return agent.environments();
+    });
+
+    return _.map(sortedEnvironments, function (environment) {
+      return new TriStateCheckbox(environment, selectedAgentsEnvironments);
+    });
+  };
+
   Environments.init = function (selectedAgents) {
     m.request({
       method:        'GET',
       url:           Routes.apiv1AdminInternalEnvironmentsPath(),
       config:        mrequest.xhrConfig.v1,
       unwrapSuccess: function (responseBody) {
-        Environments.list = _.map(responseBody, function (value) {
-          return new TriStateCheckbox(value, _.map(selectedAgents, function (agent) {
```

Comment:
Yeah. I will do ```responseBody.sort()``` but will put it in another method because ```_.map(selectedAgents, function (agent) ``` part of the code iterates for every environment which is not required. And it will be difficult to read if I extract that variable in ```unwrapSuccess``` itself.